tests/app/views/test_application.py

```diff
@@ -9,6 +9,12 @@
 from six.moves import zip as izip
 
 
+def test_approval(sample_submitted_application):
+    assert sample_submitted_application.supplier is None
+    sample_submitted_application.set_approval(True)
+    assert sample_submitted_application.supplier is not None
```

Comment:
Ideally. At the moment this is just proving that the conversion didn't fail. Lots more we can add.